scripts/launcher/lib/virtual_controller.py

```diff
@@ -145,8 +146,8 @@ def _prepare_args(self):
             extra_args += " inst.cmdline"
         if self._kernel_args:
             extra_args += " " + self._kernel_args
-        if self._iso.stage2:
-            extra_args += " stage2=hd:CDLABEL={0}".format(udev_escape(self._iso.label))
+
+        extra_args += " stage2=hd:CDLABEL={0}".format(udev_escape(self._label))
```

Comment:
Ditto here `inst.stage2=...`

Reply:
Ack, thanks for spotting! I locally ran the test, and it works fine with the change.